extensions/poetry/main.ts

```diff
@@ -146,74 +103,98 @@ Deno.exit(status.code);
 async function poetryCheckDryRun(
   subcommand: string,
   args: string[],
-): Promise<number> {
-  // Skip lockfile update on install, since it doesn't have the `--lock` flag.
-  if (subcommand !== "install") {
-    console.log(`[${green("phylum")}] Updating lockfile…`);
-
-    const status = PhylumApi.runSandboxed({
-      cmd: "poetry",
-      args: [subcommand, "-n", "--lock", ...args.map((s) => s.toString())],
-      exceptions: {
-        run: [
-          "./",
-          "/bin",
-          "/usr/bin",
-          "~/.pyenv",
-          "~/.local/bin/poetry",
-          "~/Library/Application Support/pypoetry",
-          "~/.local/share/pypoetry",
-        ],
-        write: [
-          "./",
-          "~/.cache/pypoetry",
-          "~/Library/Caches/pypoetry",
-          "~/.pyenv",
-        ],
-        read: [
-          "./",
-          "~/.cache/pypoetry",
-          "~/Library/Caches/pypoetry",
-          "~/.pyenv",
-          "~/Library/Preferences/pypoetry",
-          "~/.config/pypoetry",
-          "/etc/passwd",
-        ],
-        net: true,
-      },
-    });
-
-    // Ensure dry-run update was successful.
-    if (!status.success) {
-      console.error(`[${red("phylum")}] Lockfile update failed.\n`);
-      await abort(status.code ?? 255);
+) {
+  const result = PhylumApi.runSandboxed({
+    cmd: "poetry",
+    args: [subcommand, "-n", "--dry-run", ...args.map((s) => s.toString())],
+    exceptions: {
+      run: [
+        "./",
+        "/bin",
+        "/usr/bin",
+        "~/.pyenv",
+        "~/.local/bin/poetry",
+        "~/Library/Application Support/pypoetry",
+        "~/.local/share/pypoetry",
+      ],
+      write: [
+        "~/.cache/pypoetry",
+        "~/Library/Caches/pypoetry",
+        "~/.pyenv",
+      ],
+      read: [
+        "./",
+        "~/.cache/pypoetry",
+        "~/Library/Caches/pypoetry",
+        "~/.pyenv",
+        "~/Library/Preferences/pypoetry",
+        "~/.config/pypoetry",
+        "/etc/passwd",
+      ],
+      net: true,
+    },
+    stdout: "piped",
+  });
+
+  // Ensure dry-run update was successful.
+  if (!result.success || result.stdout.length == 0) {
+    console.error(`[${red("phylum")}] Failed to determine new packages.\n`);
+    return status.code ?? 255;
+  }
+
+  // Parse dry-run output to look for new packages.
+  const packages = [];
```

Comment:
As far as I know, const is for variables which are never reassigned. Pushing to an array is not a reassignment, so defining it as `const` is correct (and our linter would warn otherwise I believe).